### PR TITLE
New UI Dark/Light Mode

### PR DESCRIPTION
**Pull request by teradaian**
Feature: Adds basic scalable light/dark mode functionality using a wrapping Styled Component themeprovider + a MUI theme provider. 

The useDarkTheme custom hook handles the styled component wrapper, which the MUI theme pairs with. All MUI features should work with dark mode now - if not let me know.

I included a Styled Component button for the toggle, but you could of course just make this a normal react feature and limit Styled Component integration to globalStyles only.  globalStyles.js atm is using mostly simple selectors, but as things scale you can increase specificity ofc.

Let me know your thoughts on this second draft, thank you :)

### CODE REVIEW

**Comment by stephin007** [comment]
checking @teradaian , lot of changes to review please give me some time!
Also seems like rightnow the site is not responsive , what are your plans for that?

File: src/components/useDarkMode.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if (theme === 'light') {
+        setMode('dark');
+    } else {
+        setMode('light');
+    }
+};
+
+useEffect(() => {
+    const localTheme = window.localStorage.getItem('theme');
+    if (localTheme) {
```
Comment (stephin007) [inline src/components/useDarkMode.js:22]: hey @teradaian 
 local theme === ?? , is it equal to either light or dark?  what is it defaulted to?

**Comment by stephin007** [comment]
@teradaian honestly, this draft looks dope! ⭐ left some comments please look into that as well!

**Comment by teradaian** [comment]
> checking @teradaian , lot of changes to review please give me some time!
> Also seems like rightnow the site is not responsive , what are your plans for that?

I'm trying to figure out what failed- hopefully just a silly oversight on my end as it was working when I committed.

**Comment by teradaian** [comment]
> @teradaian honestly, this draft looks dope! ⭐ left some comments please look into that as well!

Thanks! I'll look into all the comments, and appreciate the feedback!

File: src/components/useDarkMode.js (hunk 1 above)
Comment (teradaian) [reply src/components/useDarkMode.js:22]: it's reading local storage and checking if 'theme' is present (as it default checks as boolean without the second argument). The idea is to make sure it stays on dark mode or light mode when selected vs resetting on reload etc.

**Review by stephin007 — APPROVED** [review]

**Comment by stephin007** [comment]
Also @teradaian the colors in the vaccine card are not suitable for the darkmode!
What are your thoughts?


Also FYI: please pull the latest changes from the NEW-UI branch to avoid conflicts

**Comment by stephin007** [comment]
Please fix the merge conflicts, @teradaian 🙏🏾

**Comment by stephin007** [comment]
@teradaian  ihave merged your PR, we will be working on the CSS issue!

Appreciate all  your efforts learnt a lot from you! looking forward to work on more issues with you!

**Comment by Justinnn07** [comment]
nice work @teradaian 💯

**Comment by teradaian** [comment]
the difference between the MUIProvider and themeprovider is that the
MUIprovider is strictly dealing with the theme setting of MUI elements,
while the styled components deals with everything else. Since they couldn't
share a name on import, I gave them more specific names.

If you check the MUI docs, you can also make changes to what the "dark"
mode consists of, to your liking. I used their default so far :)

**Pull request merged**